Springer Link.js

```diff
@@ -31,185 +29,191 @@ function detectWeb(doc, url) {
-			item.ISSN = ZU.xpathText(doc,
-				'//dd[@id="abstract-about-issn" or\
+	}
+	if(itemType == "journalArticle" && !item.ISSN) {
+		item.ISSN = ZU.xpathText(doc,
+			'//dd[@id="abstract-about-issn" or\
 					@id="abstract-about-electronic-issn"]');
-		}
-
-		if(itemType == 'bookSection') {
-			//look for editors
-			var editors = ZU.xpath(doc,
-				'//ul[@class="editors"]/li[@itemprop="editor"]\
+	}
+	if(itemType == 'bookSection') {
+		//look for editors
+		var editors = ZU.xpath(doc, '//ul[@class="editors"]/li[@itemprop="editor"]\
 					/a[@class="person"]');
-			var m = item.creators.length;
-			for(var i=0, n=editors.length; i<n; i++) {
-				var editor = ZU.cleanAuthor(
-						editors[i].textContent.replace(/\s+Ph\.?D\.?/, ''),
-						'editor');
-				//make sure we don't already have this person in the list
-				var haveEditor = false;
-				for(var j=0; j<m; j++) {
-					var creator = item.creators[j];
-					if(creator.creatorType == "editor"
-						&& creator.lastName == editor.lastName) {
-						/* we should also check first name, but this could get
-						   messy if we only have initials in one case but not
-						   the other. */
-						haveEditor = true;
-						break;
-					}
-				}
-
-				if(!haveEditor) {
-					item.creators.push(editor);
+		var m = item.creators.length;
+		for(var i = 0, n = editors.length; i < n; i++) {
+			var editor = ZU.cleanAuthor(editors[i].textContent.replace(/\s+Ph\.?D\.?/, ''), 'editor');
+			//make sure we don't already have this person in the list
+			var haveEditor = false;
+			for(var j = 0; j < m; j++) {
+				var creator = item.creators[j];
+				if(creator.creatorType == "editor" && creator.lastName == editor.lastName) {
+					/* we should also check first name, but this could get
+					   messy if we only have initials in one case but not
+					   the other. */
+					haveEditor = true;
+					break;
 				}
 			}
-
-			if(!item.ISBN) {
-				item.ISBN = ZU.xpathText(doc,
-					'//dd[@id="abstract-about-book-print-isbn" or\
-						@id="abstract-about-book-online-isbn"]');
-			}
-
-			//series/seriesNumber
-			if(!item.series) {
-				item.series = ZU.xpathText(doc,
-					'//dd[@id="abstract-about-book-series-title"]');
-			}
-
-			if(!item.seriesNumber) {
-				item.seriesNumber = ZU.xpathText(doc,
-					'//dd[@id="abstract-about-book-series-volume"]');
+			if(!haveEditor) {
+				item.creators.push(editor);
 			}
 		}
+		if(!item.ISBN) {
+			item.ISBN = ZU.xpathText(doc,
+				'//dd[@id="abstract-about-book-print-isbn" or\
+						@id="abstract-about-book-online-isbn"]');
+		}
+		//series/seriesNumber
+		if(!item.series) {
+			item.series = ZU.xpathText(doc, '//dd[@id="abstract-about-book-series-title"]');
+		}
+		if(!item.seriesNumber) {
+			item.seriesNumber = ZU.xpathText(doc, '//dd[@id="abstract-about-book-series-volume"]');
+		}
+	}
+	//add the DOI to extra for non journal articles
+	if(item.itemType != "journalArticle" && item.DOI) {
+		item.extra = "DOI: " + item.DOI;
+	}
```

Comment:
Not right now, but maybe we can think about doing it. For now, delete `item.DOI` after moving it to `item.extra` so this doesn't break in the future.